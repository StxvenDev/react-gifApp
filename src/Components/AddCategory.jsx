import React from 'react'
import { useState } from "react"
import PropTypes from 'prop-types'
export const AddCategory = ({onNewValue}) => {
  const [inputValue, setInputValue] = useState('');
  const onInputChange = ({target}) => {
    setInputValue(target.value);
  }
  const onSubmit = (event) => {
    const inputTrim = inputValue.trim();
    event.preventDefault();
    if(inputTrim.length <= 1 ) return;
    // setCategories(c => [inputValue, ...c]);
    onNewValue(inputTrim);
    setInputValue('');
  }
  return (
    <form onSubmit={(event) => onSubmit(event)} aria-label='form'>
      <input 
          type="text"
          placeholder="Buscar gifs"
          value={inputValue} 
          onChange={onInputChange}
      />
    </form>
  )
}

AddCategory.propTypes = {
  onNewValue : PropTypes.func.isRequired
}
