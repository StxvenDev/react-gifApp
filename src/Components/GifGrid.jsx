import { GifItem } from "./GifItem";
import { useFetchGifs } from "../helpers/useFetchGifs";
export const GifGrid = ({ category }) => {
  const {images, isLoading} = useFetchGifs(category);
  return (
    <>
      <h3>{category}</h3>
      <div className="card-grid">
        {
          isLoading && (<h2>Cargando...</h2>)
        }
        {
          images.map((image) => (
            <GifItem 
              key={image.id} 
              {...image} 
            />
          ))
        }
      </div>
    </>
  );
};
