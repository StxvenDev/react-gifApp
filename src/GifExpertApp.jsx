import React from 'react'
import {useState} from 'react'
import { AddCategory, GifGrid } from './Components';

export const GifExpertApp = () => {
  const [Categories, setCategories] = useState(['shingeki no kyojin', 'naruto']);
  const onAddCategory = (newCategory = '') => {
    const lowerCategories = Categories.map(category => category.toLowerCase());
    const lowerNewCategory = newCategory.toLowerCase();
    if(lowerCategories.includes(lowerNewCategory)) return;
    setCategories([newCategory, ...Categories]);
  };
  return (
    <>
      <h1>GifExpertApp</h1>
      <AddCategory 
        // setCategories={setCategories} 
        onNewValue={value => onAddCategory(value)}
      />
      {
        Categories.map((category) =>(
          <GifGrid key={category} category={category} />
        ))
      }
    </>
  )
}
