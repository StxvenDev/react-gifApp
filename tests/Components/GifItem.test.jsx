import React from "react"
import { render, screen } from "@testing-library/react"
import {GifItem} from '../../src/Components' 

describe('Pruebas en GifItem', () => { 
  const title = 'Sukuna';
  const url = 'https://sukuna-backflash.org/';
  test('It should match snapshot', () => { 
    const {container} = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
   });
   test('Url y title deben ser iguales', () => { 
      render(<GifItem title={title} url={url} />)
      const {src, alt} = screen.getByRole('img');
      expect(src).toBe(url);
      expect(alt).toBe(title);
    });
    test('debe de mostrar el titulo en el componente', () => { 
      render(<GifItem title={title} url={url} />);
      expect(screen.getByText(title)).toBeTruthy();
     })
 })