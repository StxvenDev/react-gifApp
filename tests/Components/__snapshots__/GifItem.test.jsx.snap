// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem It should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Sukuna"
      src="https://sukuna-backflash.org/"
    />
    <p>
      Sukuna
    </p>
  </div>
</div>
`;
